### ISSUE TITLE
Django project can't find Pillow

### ISSUE THREAD

**Issue by liuchanganshan** [python-pillow/Pillow#5212]
MacBook Pro  (13inch M1chip) can't install Pillow,(Python environment is Python 3.9.1rc1 (v3.9.1rc1:88db374422, Nov 24 2020, 17:00:47) )

**Comment by liuchanganshan** [comment]
I tried again ,using this method   https://blog.csdn.net/jia611/article/details/111415011, make it, install Pillow successfully via my virtulenv pip3 install.
However, When I start my Django project ,it didn't work.
this is the error info output:


Watching for file changes with StatReloader
Performing system checks...

Exception in thread django-main-thread:
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/threading.py", line 954, in _bootstrap_inner
    self.run()
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/threading.py", line 892, in run
    self._target(*self._args, **self._kwargs)
  File "{{MyDjangoProjectName}}/venv/lib/python3.9/site-packages/django/utils/autoreload.py", line 54, in wrapper
    fn(*args, **kwargs)
  File "{{MyDjangoProjectName}}/venv/lib/python3.9/site-packages/django/core/management/commands/runserver.py", line 117, in inner_run
    self.check(display_num_errors=True)
  File "{{MyDjangoProjectName}}/venv/lib/python3.9/site-packages/django/core/management/base.py", line 436, in check
    raise SystemCheckError(msg)
django.core.management.base.SystemCheckError: SystemCheckError: System check identified some issues:

ERRORS:
equipment_account.EquipmentInfo.img: (fields.E210) Cannot use ImageField because Pillow is not installed.
	HINT: Get Pillow at https://pypi.org/project/Pillow/ or run command "pip install Pillow".
equipment_account.EquipmentInfo.qrcode: (fields.E210) Cannot use ImageField because Pillow is not installed.
	HINT: Get Pillow at https://pypi.org/project/Pillow/ or run command "pip install Pillow".
equipment_account.Station.img: (fields.E210) Cannot use ImageField because Pillow is not installed.
	HINT: Get Pillow at https://pypi.org/project/Pillow/ or run command "pip install Pillow".
equipment_account.Station.qrcode: (fields.E210) Cannot use ImageField because Pillow is not installed.
	HINT: Get Pillow at https://pypi.org/project/Pillow/ or run command "pip install Pillow".
no_equipment_blog.NoEquipmentBlog.img: (fields.E210) Cannot use ImageField because Pillow is not installed.
	HINT: Get Pillow at https://pypi.org/project/Pillow/ or run command "pip install Pillow".
users.UserProfile.image: (fields.E210) Cannot use ImageField because Pillow is not installed.
	HINT: Get Pillow at https://pypi.org/project/Pillow/ or run command "pip install Pillow".
worklog.WorkLog.img: (fields.E210) Cannot use ImageField because Pillow is not installed.
	HINT: Get Pillow at https://pypi.org/project/Pillow/ or run command "pip install Pillow".

System check identified 7 issues (0 silenced).


Pillow can be installed, but can't be used in Django.

**Comment by radarhere** [comment]
Hi. From your Django code, could you try `import PIL` by itself and let us know if that works or not?

**Comment by radarhere** [comment]
And actually the output from `print(PIL)` after that could also be interesting, to show path where your Pillow installation is located.

**Comment by liuchanganshan** [comment]
@radarhere 
I've tried ,error info "Cannot use ImageField because Pillow is not installed.
	HINT: Get Pillow at https://pypi.org/project/Pillow/ or run command "pip install Pillow"." is thrown by Django/db/models/fields 
class ImageField(FileField):
    def _check_image_library_installed(self):
        try:
            from PIL import Image  # NOQA
        except ImportError:
            return [
                checks.Error(
                    'Cannot use ImageField because Pillow is not installed.',
                    hint=('Get Pillow at https://pypi.org/project/Pillow/ '
                          'or run command "pip install Pillow".'),
                    obj=self,
                    id='fields.E210',
                )
            ]
        else:
            return []
I've tried to delete try except block,use  "from PIL import Image"straightly, but the error didin't disappear.

**Comment by radarhere** [comment]
Ok, how about `pip show Pillow` so we can see the install path that way?

**Comment by liuchanganshan** [comment]
pip show Pillow
Name: Pillow
Version: 8.1.0
Summary: Python Imaging Library (Fork)
Home-page: https://python-pillow.org
Author: Alex Clark (PIL Fork Author)
Author-email: aclark@python-pillow.org
License: HPND
Location: $ProjectDirectory/venv/lib/python3.9/site-packages
Requires: 
Required-by:

**Comment by liuchanganshan** [comment]
error still exist

**Comment by radarhere** [comment]
I'm grasping at straws a bit, but my theory is (still) that Pillow and Django are just installed in different environments - what does `pip show Django` give?

**Comment by radarhere** [comment]
I've renamed this issue, since we've moved on from the problem with installation on an M1.

**Comment by liuchanganshan** [comment]
@radarhere 
you can attempt to install the python3.9.1 x64 installer pkg ,and the problem has been resolved temporarily.
the problem came from 3.9.1 arm version python, and make the Django package can't find Pillow.

**Comment by radarhere** [comment]
I don't believe it should be purely because of the installer you used. I think it is going to be because of some different way that you installed Pillow and Django.

Feel free to try and demonstrate that I'm wrong though - could you list out all the commands that you run to install Django and Pillow?

Or if it's easier for the moment, could you let us know what `pip show Django` gives?

**Comment by Cellou404** [comment]
I've the problem

**Comment by Cellou404** [comment]
I've the same problem

**Comment by radarhere** [comment]
Django can't find Pillow? Please provide some more information.

**Comment by xiongruixiang** [comment]
@radarhere I have the same problem. Can you give me some support?

$ conda list pillow
# packages in environment at /Users/opt/anaconda3/envs/rango:
#
# Name                    Version                   Build  Channel
pillow                    5.4.1            py37hb68e598_0


`class UserProfile(models.Model):
    user = models.OneToOneField(User, on_delete=models.CASCADE)
    website = models.URLField(blank=True)
    picture = models.ImageField(upload_to='profile_images', blank=True)

    def __str__(self):
        return self.user.username`


Below is the error message.

$ python manage.py makemigrations rango
SystemCheckError: System check identified some issues:

ERRORS:
rango.UserProfile.picture: (fields.E210) Cannot use ImageField because Pillow is not installed.
        HINT: Get Pillow at https://pypi.org/project/Pillow/ or run command "pip install Pillow".

**Comment by radarhere** [comment]
I shouldn't claim any familiarity with conda, but if you're happy with a `pip` solution, have you tried `python -m pip install Pillow`?

I doubt your problem is related to any of the code within Pillow, so you could try solutions for other Python packages not being detected by conda - https://stackoverflow.com/questions/46767012/conda-list-shows-a-package-but-cannot-import-it for example.

As an aside, I should recommend that you upgrade from Pillow 5.4.1.

**Comment by LetsGoetz** [comment]
Had same issue, now solved, here is what happend:

`
PS D:\Code\py\generic_webshop> py manage.py makemigrations

SystemCheckError: System check identified some issues:
ERRORS:
webshop_base.Customer.avatar_image: (fields.E210) Cannot use ImageField because Pillow is not installed.
        HINT: Get Pillow at https://pypi.org/project/Pillow/ or run command "python -m pip install Pillow".
webshop_base.ProductCategory.category_image: (fields.E210) Cannot use ImageField because Pillow is not installed.
        HINT: Get Pillow at https://pypi.org/project/Pillow/ or run command "python -m pip install Pillow".
webshop_base.ProductImage.product_image: (fields.E210) Cannot use ImageField because Pillow is not installed.
        HINT: Get Pillow at https://pypi.org/project/Pillow/ or run command "python -m pip install Pillow".
webshop_base.ProductReview.review_image: (fields.E210) Cannot use ImageField because Pillow is not installed.
        HINT: Get Pillow at https://pypi.org/project/Pillow/ or run command "python -m pip install Pillow".


PS D:\Code\py\generic_webshop> pip show Pillow

Name: Pillow
Version: 8.3.1
Summary: Python Imaging Library (Fork)
Home-page: https://python-pillow.org
Author: Alex Clark (PIL Fork Author)
Author-email: aclark@python-pillow.org
License: HPND
Location: c:\python39\lib\site-packages
Requires:
Required-by:


PS D:\Code\py\generic_webshop> pip show django

Name: Django
Version: 3.2.6
Summary: A high-level Python Web framework that encourages rapid development and clean, pragmatic design.
Home-page: https://www.djangoproject.com/
Author: Django Software Foundation
Author-email: foundation@djangoproject.com
License: BSD-3-Clause
Location: c:\python39\lib\site-packages
Requires: sqlparse, asgiref, pytz
Required-by:


PS D:\Code\py\generic_webshop> python -m pip install --upgrade pip

Collecting pip
  Using cached pip-21.2.4-py3-none-any.whl (1.6 MB)
Installing collected packages: pip
  Attempting uninstall: pip
    Found existing installation: pip 20.2.3
    Uninstalling pip-20.2.3:
      Successfully uninstalled pip-20.2.3
Successfully installed pip-21.2.4


PS D:\Code\py\generic_webshop> pip install --force-reinstall Pillow

WARNING: Ignoring invalid distribution -ip (c:\python39\lib\site-packages)
WARNING: Ignoring invalid distribution -ip (c:\python39\lib\site-packages)
Collecting Pillow
  Downloading Pillow-8.3.2-cp39-cp39-win_amd64.whl (3.2 MB)
     |████████████████████████████████| 3.2 MB 1.6 MB/s
WARNING: Ignoring invalid distribution -ip (c:\python39\lib\site-packages)
Installing collected packages: Pillow
  Attempting uninstall: Pillow
    WARNING: Ignoring invalid distribution -ip (c:\python39\lib\site-packages)
    Found existing installation: Pillow 8.3.1
    Uninstalling Pillow-8.3.1:
      Successfully uninstalled Pillow-8.3.1
WARNING: Ignoring invalid distribution -ip (c:\python39\lib\site-packages)
Successfully installed Pillow-8.3.2
WARNING: Ignoring invalid distribution -ip (c:\python39\lib\site-packages)
WARNING: Ignoring invalid distribution -ip (c:\python39\lib\site-packages)
PS D:\Code\py\generic_webshop> pip show Pillow
WARNING: Ignoring invalid distribution -ip (c:\python39\lib\site-packages)
Name: Pillow
Version: 8.3.2
Summary: Python Imaging Library (Fork)
Home-page: https://python-pillow.org
Author: Alex Clark (PIL Fork Author)
Author-email: aclark@python-pillow.org
License: HPND
Location: c:\python39\lib\site-packages
Requires:
Required-by:


PS D:\Code\py\generic_webshop> py manage.py makemigrations
...
Works now!
`

**Comment by shadyueh** [comment]
Dude, I've checked and Pillow was installed by requirements. I forced the reinstall, but no luck. Django by some reason still can't find Pillow. Mine is in 8.4.0

**Comment by radarhere** [comment]
Are you running Django and installing Pillow in to the same Python version? Could you show the commands that you are using?

**Comment by shubhajeet1207** [comment]
> pip install --force-reinstall Pillow

This solution works for me. Thanx buddy.

**Comment by Parash-shresh** [comment]
i solved this error by : 
copy paste the entire PIL foler to the virtual environment folder of django
C:\Python310\Lib\site-packages\PIL copy this to C:\Users\xxx\Desktop\project\venv\Lib\site-packages

**Comment by alirezabhr** [comment]
I had this problem on macbook pro (M1 pro chip).
reinstalling Pillow works for me.
`pip install --force-reinstall Pillow`

**Comment by yuis-ice** [comment]
'key, why closing? I'm on ubuntu 20.04 **not** mac osx. 

my env: 

```
$ pip freeze|ag django=
Django==3.1.7

$ pip freeze|ag pillow=
Pillow==9.0.1

$ python --version
Python 3.7.6

$ pip --version
pip 21.2.4 from /home/yuis/pg/django_v2/source/venv_linux/lib/python3.7/site-packages/pip (python 3.7)
```

uninstalling and reinstalling didn't help. Tried to install a different version (5.0.0) installing deps `apt install libjpeg-dev zlib1g-dev`, this also didn't help. 

anyone knows a workaround to `models.ImageField()`? I didn't expect stcuking in this.

**Comment by hugovk** [comment]
Please could you open a new issue?

Tip: don't run pip directly.

Short version: https://twitter.com/nedbat/status/1501564714442280967

![image](https://user-images.githubusercontent.com/1324225/158332464-75c65988-83ec-457c-9752-98b7be39ff7d.png)

Long version: https://snarky.ca/why-you-should-use-python-m-pip/
